### PR TITLE
Add OpenAPI support

### PR DESCRIPTION
**Pull request by angoglez**
### **Description**
Related issue (marlow): https://github.com/47deg/marlow/issues/636
Related issue (compendium): https://github.com/higherkindness/compendium/issues/30
This PR adds compendiun OpenAPI support (using YAML or JSON files)

### **Found issues**

- When storing a schema specifying its idl type it's `openapiyaml` but you insert as raw a correct OpenAPI JSON, validation step it's done correctly and the schema it's stored. This has something to do with Skeumorph, thought it provides different parsers for each type of file. I think this can lead to inconsistency but I don't have ideas about how to prevent it. It doesn't happen the other way round. (See https://github.com/higherkindness/skeuomorph/issues/303)

### CODE REVIEW

**Comment by fedefernandez** [comment]
@angoglez it seems this has problems with the scalafmt format and compilation errors. Could you please re-format it with `scalafmtAll` and then fix the compilation errors. I'd be happy to review once it's green 📗

File: src/main/resources/db/migration/metadata/V1__add_metaprotocols_table.sql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CREATE TYPE idl AS ENUM ('avro', 'protobuf', 'mu', 'openapi', 'scala');
+CREATE TYPE idl AS ENUM ('avro', 'protobuf', 'mu', 'openapiyaml', 'openapijson', 'scala');
```
Comment (fedefernandez) [inline src/main/resources/db/migration/metadata/V1__add_metaprotocols_table.sql:1]: We shouldn't change migrations. If we need a change we need to create a new migration file. The idea with the migrations is to manage DB evolutions and changing a previous one makes for an existing installation would make it fail.

I think [this post](https://blog.yo1.dog/updating-enum-values-in-postgresql-the-safe-and-easy-way/) covers what we need here, specially in the `Updating/Renaming a Value`

File: src/main/resources/db/migration/metadata/V2__update_idl_types.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+ALTER TYPE idl ADD VALUE 'openapiyaml';
+ALTER TYPE idl ADD VALUE 'openapijson';
```
Comment (fedefernandez) [inline src/main/resources/db/migration/metadata/V2__update_idl_types.sql:2]: We don't need this

```suggestion
```

File: src/main/resources/db/migration/metadata/V2__update_idl_types.sql [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+UPDATE metaprotocols SET idl_name = 'openapiyaml' WHERE idl_name = 'openapi'
```
Comment (fedefernandez) [inline src/main/resources/db/migration/metadata/V2__update_idl_types.sql:4]:
```suggestion
UPDATE metaprotocols SET idl_name = 'openapiyaml' WHERE idl_name = 'openapi';
```

File: src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala [hunk 4]
```diff
@@ -62,27 +63,97 @@ object ProtocolUtilsSpec extends Specification with ScalaCheck {
...
+      validator
+        .validateProtocol(protocol, IdlName.OpenAPI)
+        .unsafeRunSync === protocol
+    }
+
+    "[OpenAPI - Yaml] Given a raw protocol text raises an error if the protocol is incorrect" >> {
+      val stream: InputStream = getClass.getResourceAsStream("/wrongYamlOpenAPI.yaml")
+      val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString("\n")
+      val protocol: Protocol  = Protocol(text)
+      stream.close()
```
Comment (juanpedromoreno) [inline src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala:108]: **suggestion**: use [Resource](https://typelevel.org/cats-effect/datatypes/resource.html#with-scalaiosource).

It doesn't need to be part of this PR :)

File: src/main/resources/db/migration/metadata/V2__update_idl_types.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+ALTER TYPE idl RENAME TO idl_old;
+
+CREATE TYPE idl as ENUM('avro', 'protobuf', 'mu', 'openapi', 'openapijson', 'scala');
+
+ALTER TABLE metaprotocols ALTER COLUMN idl_name TYPE idl USING idl_name::text::idl;
+
+DROP TYPE idl_old;
```
Comment (fedefernandez) [inline src/main/resources/db/migration/metadata/V2__update_idl_types.sql:7]: Rethinking about this, we could just have the following:
```suggestion
ALTER TYPE idl ADD VALUE 'openapijson';
```

File: src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala (hunk 4 above)
Comment (fedefernandez) [reply src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala:108]: Good idea!, but as you said I'd address it in a following PR

Comment (angoglez) [reply src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala:108]: Sure, thanks @juanpedromoreno! We'll have this in mind.

**Pull request merged**